### PR TITLE
Add adapter path support for cloud model cache

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add adapter path support for cloud model cache

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

**Comment by xiaoyu-work** [comment]
Will update this RP in this week.

File: olive/engine/cloud_cache_helper.py [hunk 1]
```diff
@@ -46,40 +45,61 @@ def __init__(
...
+            model_config.config["adapter_path"] = str(output_model_path / adapter_path)
+
+        return model_config
+
+    def download_blob_list(self, blob_list, directory_prefix, output_model_path, prefix: str = None):
         for blob in blob_list:
             blob_client = self.container_client.get_blob_client(blob)
-            local_file_path = self.output_model_path / blob.name[len(model_directory_prefix) + 1 :]
+            local_file_path = (
+                output_model_path / prefix / blob.name[len(directory_prefix) + 1 :]
```
Comment (jambayk) [inline olive/engine/cloud_cache_helper.py:75]: instead of doing string slice, maybe you could do something like `Path(blob.name).name`?

Comment (xiaoyu-work) [reply olive/engine/cloud_cache_helper.py:75]: `Path(blob.name).name` will lose the subfolder structure of this blob. For example if I have `prefix/subfolder/file.name`, `Path(blob.name).name` will only get `file.name`.

Comment (jambayk) [reply olive/engine/cloud_cache_helper.py:75]: Oh I see.

File: olive/engine/cloud_cache_helper.py [hunk 2]
```diff
@@ -29,7 +29,6 @@ class CloudCacheConfig(ConfigBase):
 class CloudCacheHelper:
```
Comment (jambayk) [inline olive/engine/cloud_cache_helper.py:29]: While you are updating this could you also add some docstrings for the methods and comments about the cloud cache helpers? Otherwise, it's hard to follow the logic if this needs to be updated as part of other changes.

Comment (xiaoyu-work) [reply olive/engine/cloud_cache_helper.py:29]: Added
